apps/desks.py

```diff
@@ -321,6 +322,36 @@ def get_desk_name(self, desk_id):
 
         return desk_name
 
+    def get(self, req, lookup):
+        desks = list(super().get(req, lookup))
+
+        members_list = []
+        db_users = app.data.mongo.pymongo('users').db['users']
+
+        # find display_name from the users document for each member in desks document
+        for desk in desks:
+            if 'members' in desk:
+                users = list(db_users.find({
+                    '_id': {'$in': [member['user'] for member in desk['members']]}},
+                    {'display_name': 1}
+                ))
+
+                for user in users:
+                    if not any(item == user for item in members_list):
+                        members_list.append(user)
+
+        if members_list:
+            members_list.sort(key=lambda k: k['display_name'].lower())
+            ordered_dict = []
+            for member in members_list:
+                ordered_dict.append(member['_id'])
+
+            # sort the members of each desk according to ordered_dict
+            for desk in desks:
+                desk['members'].sort(key=lambda x: ordered_dict.index(x['user']))
```

Comment:
here you should probably check again `if 'members' in desk:`

Reply:
I don't think so, as I already have this check

```suggestion
        if members_list:
```
and `members_list` will be non-empty only if there are `'members' in desk` which I have already checked

Reply:
but as long as there is single desk with members it will populate members_list, but there could be still other desks without any members.

Reply:
Ok, I see I have updated it.